### ISSUE TITLE
Strategy for OAuth Token Migration v1 to v2

### ISSUE THREAD

**Issue by tobihagemann** [dropbox/dropbox-sdk-obj-c#117]
How does `+[DBClientsManager checkAndPerformV1TokenMigration:queue:appKey:appSecret:]` fit into the setup process? When should I call `+[DBClientsManager setupWithAppKey:]` (or any other equivalent setup method)?

The thing is: I can only call the setup method once. If I wanted to check and perform a token migration, I'd have to call the setup method AFTER the migration. But how do I check if a migration is even needed? I can see here [DBSDKKeychain.m#L156](https://github.com/dropbox/dropbox-sdk-obj-c/blob/f0aa8518cea20103756f2c24e773388cf2ff6d21/Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBSDKKeychain.m#L156) how it's checked, but that's not exposed publicly.

This is what I'm currently doing:

```objc
[DBClientsManager checkAndPerformV1TokenMigration:^(BOOL shouldRetry, BOOL invalidAppKeyOrSecret, NSArray<NSArray<NSString *> *> *unsuccessfullyMigratedTokenData) {
  [DBClientsManager setupWithAppKey:@"…"]];
  // calling another callback, telling my app that Dropbox setup is done
} queue:nil appKey:@"…" appSecret:@"…"];
```

It's not working due to `responseBlock` not being called, if migration hasn't occurred and there is nothing to migrate. But it works, if at least one v1 token was found for migration or migration has already occurred.

I can't call the setup method outside of `checkAndPerformV1TokenMigration`, because I want to make sure that a token has been properly migrated first. As I've mentioned, I can't call the setup method twice. What's the strategy, am I missing something obvious?

Ideas to solve this issue:
1. Add an else case for [DBSDKKeychain.m#L176](https://github.com/dropbox/dropbox-sdk-obj-c/blob/f0aa8518cea20103756f2c24e773388cf2ff6d21/Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBSDKKeychain.m#L176), so that the `responseBlock` is always called. (Edit: Created PR, before I had the second idea)
2. Add a boolean return value to `checkAndPerformV1TokenMigration`, which does the "checking" part.

**Comment by tobihagemann** [comment]
For the time being, I've created a PR, because I'd like to release a new version of my project. Not sure if that's the right solution to this issue.

**Comment by scobbe** [comment]
@tobihagemann: Thanks so much for your thoughtful feedback, as well as for making the pull request. I just merged your second solution (which seems like the best way to go).

These changes will be distributed in the latest version of the SDK, which will go out today.

Please let us know if you spot other issues with the SDK. We really appreciate it! Have a great day 😄

**Comment by scobbe** [comment]
Ok, this should be fixed in [3.0.7](https://github.com/dropbox/dropbox-sdk-obj-c/releases/tag/3.0.7) now.
